### PR TITLE
feat: use SDK v2

### PR DESCRIPTION
**Pull request by KrisQ**
## Why are these changes introduced?

Addresses [ENG-8503](https://nacelle.atlassian.net/browse/ENG-8503) 

## What is this pull request doing?

- Update the Next reference store to make use of the SDK v2

## How to Test

- `npm run build && npm run start` 
- Explore website

## Checklist

- [x] This Pull Request aligns with `nacelle-js`' [Code of Conduct](https://github.com/getnacelle/nacelle-js/blob/main/CODE_OF_CONDUCT.md#code-of-conduct).
- [x] You can follow your own "How to Test" instructions and get the expected result.
- [x] Screenshots, videos, etc. in the PR description are free of brand names and sensitive details.
- [ ] Created a [changeset](https://github.com/changesets/changesets) (if the change should appear in a changelog).
- [ ] Updated the `README.md` with documentation changes (if applicable).

[ENG-8503]: https://nacelle.atlassian.net/browse/ENG-8503?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference-stores/next/components/Collection/CollectionGrid.js
Comment (NWRichmond) [inline reference-stores/next/components/Collection/CollectionGrid.js:1]: There are issues here.

1. The `variables.handle` access should still be `collection.content?.handle`.

```js
variables: { handle: collection.content?.handle, after }
```

2. The `data` access should happen in the `collectionProducts` assignment on line 35:
```js
const collectionProducts =
  data.collections.edges[0].node?.products?.edges.map(
    (product) => product.node
  );
```

File: reference-stores/next/pages/_app.js [hunk 1]
```diff
@@ -28,12 +28,14 @@ const MyApp = ({ Component, pageProps, products, components }) => {
...
 
 MyApp.getInitialProps = async (appContext) => {
-  const data = await nacelleClient.query({ query: SITE_QUERY }).then((site) => {
-    return resolveSiteData({ client: nacelleClient, site });
-  });
+  const { data } = await nacelleClient
+    .query({ query: SITE_QUERY })
+    .then((site) => {
+      return resolveSiteData({ client: nacelleClient, site });
+    });
```
Comment (NWRichmond) [inline reference-stores/next/pages/_app.js:35]: An implicit return might be nice here

```suggestion
    .then((site) => resolveSiteData({ client: nacelleClient, site }));
```

File: reference-stores/next/pages/collections/[handle].js [hunk 2]
```diff
@@ -50,11 +50,11 @@ export async function getStaticProps({ params: { handle } }) {
   });
   const { page } = await resolvePageData({
     client: nacelleClient,
-    page: pages.edges[0]?.node
+    page: data.pages.edges[0]?.node
```
Comment (NWRichmond) [inline reference-stores/next/pages/collections/[handle].js:53]: Might be a good opportunity to use optional chaining with `.at`:

```suggestion
    page: data.pages.edges.at(0)?.node
```

File: reference-stores/next/pages/collections/[handle].js [hunk 3]
```diff
@@ -50,11 +50,11 @@ export async function getStaticProps({ params: { handle } }) {
...
   });
   return {
     props: {
-      collection: collections.edges[0].node || null,
+      collection: data.collections.edges[0].node || null,
```
Comment (NWRichmond) [inline reference-stores/next/pages/collections/[handle].js:57]: Might be a good opportunity to use optional chaining with `.at`:

```suggestion
      collection: data.collections.edges.at(0)?.node || null,
```

File: reference-stores/next/pages/index.js [hunk 4]
```diff
@@ -19,14 +19,16 @@ const Home = ({ page }) => {
...
-  const { pages } = await nacelleClient.query({
+  const { data } = await nacelleClient.query({
     query: CONTENT_PAGE_QUERY,
     variables: { handle: 'page-homepage' }
   });
+
   const { page } = await resolvePageData({
     client: nacelleClient,
-    page: pages.edges[0]?.node
+    page: data.pages.edges[0]?.node
```
Comment (NWRichmond) [inline reference-stores/next/pages/index.js:29]: Might be a good opportunity to use optional chaining with `.at`:

```suggestion
    page: data.pages.edges.at(0)?.node
```

File: reference-stores/next/pages/pages/[handle].js [hunk 5]
```diff
@@ -35,13 +35,13 @@ export async function getStaticPaths() {
...
 export async function getStaticProps({ params }) {
-  const { pages } = await nacelleClient.query({
+  const { data } = await nacelleClient.query({
     query: CONTENT_PAGE_QUERY,
     variables: { handle: `page-${params.handle}` }
   });
   const { page } = await resolvePageData({
     client: nacelleClient,
-    page: pages.edges[0]?.node
+    page: data.pages.edges[0]?.node
```
Comment (NWRichmond) [inline reference-stores/next/pages/pages/[handle].js:44]: Might be a good opportunity to use optional chaining with `.at`:

```suggestion
    page: data.pages.edges.at(0)?.node
```

File: reference-stores/next/pages/products/[handle].js [hunk 6]
```diff
@@ -48,11 +48,11 @@ export async function getStaticProps({ params: { handle } }) {
   });
   const { page } = await resolvePageData({
     client: nacelleClient,
-    page: pages.edges[0]?.node
+    page: data.pages.edges[0]?.node
```
Comment (NWRichmond) [inline reference-stores/next/pages/products/[handle].js:51]: Might be a good opportunity to use optional chaining with `.at`:


```suggestion
    page: data.pages.edges.at(0)?.node
```

File: reference-stores/next/pages/products/[handle].js [hunk 7]
```diff
@@ -48,11 +48,11 @@ export async function getStaticProps({ params: { handle } }) {
...
   });
   return {
     props: {
-      product: products.edges[0].node || null,
+      product: data.products.edges[0].node || null,
```
Comment (NWRichmond) [inline reference-stores/next/pages/products/[handle].js:55]: Might be a good opportunity to use optional chaining with `.at`:

```suggestion
      product: data.products.edges.at(0)?.node || null,
```

**Review by NWRichmond — CHANGES_REQUESTED** [review]
Feel free to ignore the optional chaining suggestions, which aren't really in the scope of this PR.

The reason I'm requesting changes is because of the issues on `reference-stores/next/components/Collection/CollectionGrid.js`. Other than that, this PR looks ready!

**Comment by KrisQ** [comment]
Thank you @NWRichmond, suggestions and change requests were applied!

**Review by NWRichmond — APPROVED** [review]
Thanks for those updates, @KrisQ!

[nacelle-next-reference-store-git-eng-8503-nextjs-766005-nacelle.vercel.app](https://nacelle-next-reference-store-git-eng-8503-nextjs-766005-nacelle.vercel.app) is working as expected.

**Pull request merged**